ARCMWDT: Some of POSIX tests are broken

**Describe the bug**
POSIX-compatible tests are broken for ARCMWDT compiler. Some types are not presented in MW library, some types mismatch with defined in Zephyr.

**To Reproduce**
Steps to reproduce the behavior:
1. ``` export ZEPHYR_TOOLCHAIN_VARIANT=arcmwdt ```
2. ``` west build -p -b nsim/nsim_em tests/posix/headers/ -T portability.posix.headers.arcmwdtlib.without_posix_api ```
4. See error

**Expected behavior**
Clean build with no errors


**Logs and console output**
```
/home/user/sandbox/zephyrproject/zephyr/include/zephyr/posix/posix_types.h:24:13: error: typedef redefinition with different types ('int' vs 'long')
   24 | typedef int pid_t;
      |             ^
/home/user/ARC_202406rc1/MetaWare/arc/inc/sys/types.h:63:19: note: previous definition is here
   63 |     typedef     long    pid_t;
      |                         ^
1 error generated.
```
and
```
/home/user/sandbox/zephyrproject/zephyr/include/zephyr/posix/aio.h:24:18: error: field has incomplete type 'struct sigevent'
   24 |         struct sigevent aio_sigevent;
      |                         ^
/home/user/sandbox/zephyrproject/zephyr/include/zephyr/posix/aio.h:24:9: note: forward declaration of 'struct sigevent'
   24 |         struct sigevent aio_sigevent;
      |                ^
1 error generated.
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: arcmwdt
 - Zephyr version: 3.7.0-rc1

--- a/include/zephyr/posix/posix_types.h
+++ b/include/zephyr/posix/posix_types.h
@@ -21,7 +21,9 @@
 extern "C" {
 #endif
 
+#if !defined(CONFIG_ARCMWDT_LIBC)
 typedef int pid_t;
+#endif
 
 #ifndef __useconds_t_defined
 typedef unsigned long useconds_t;

--- a/lib/libc/arcmwdt/include/signal.h
+++ b/lib/libc/arcmwdt/include/signal.h
@@ -1,0 +1,12 @@
+/*
+ * Copyright (c) 2024 Synopsys
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIB_LIBC_ARCMWDT_INCLUDE_SIGNAL_H_
+#define LIB_LIBC_ARCMWDT_INCLUDE_SIGNAL_H_
+
+#include <zephyr/posix/signal.h>
+
+#endif /* LIB_LIBC_ARCMWDT_INCLUDE_SIGNAL_H_ */

